### ISSUE TITLE
Xamarin Mobile Solution Won't Build VS2017 - Android.Support.Vector.Drawable

### ISSUE THREAD

**Issue by aherrick** [dotnet/eShopOnContainers#98]
I've tried restoring Packages folder receiving the same errors.

This is a fresh install of VS2017 and pull of latest code. Also why the errors in Spanish?

Seems to be having an issue with:

`\packages\Xamarin.Android.Support.Vector.Drawable.23.3.0\build\Xamarin.Android.Support.Vector.Drawable.targets
`

But I verified that file exists.

![3-8-2017 7-49-09 am](https://cloud.githubusercontent.com/assets/586671/23704745/55d39116-03d4-11e7-9725-17f48e8f8871.png)

**Comment by aherrick** [comment]
Sounds good thanks. Is there anything I can do in the meantime to get around this?

**Comment by CESARDELATORRE** [comment]
We're actively looking what can be happening to your environment. In the tests we made the Xamarin app is working good. We'll test in additional environments.
We'll come back to this issue as soon we have further news, ok?
Thanks,

**Comment by CESARDELATORRE** [comment]
@aherrick You need to have installed API Level 23, Android 6.0 from the Android SDK. However, it should have been checked and installed when opening the project. We'll research that.

As an initial solution, can you go and install the API Level 23, Android 6.0 from the Android SDK, as below?
![android_api-level_23_needed](https://cloud.githubusercontent.com/assets/1712635/23723055/8dc27678-03fc-11e7-961e-74bf6ad353ab.png)

Then, tell us if that solved your issue.

In regards the message in Spanish, it is not from our team but from VS.
Basically, the iOS and Android projects need to have certain specific packages. In the case of Android, this is the error/check:

![error_spanish_api-level_needed](https://cloud.githubusercontent.com/assets/1712635/23722903/f8efc712-03fb-11e7-8bc3-6cc7203b94a7.png)

The fact that the error was stored in Spanish, by Visual Studio, within the .CSPROJ is because the VS version used to create the Xamarin project was a VS Spanish. We'll fix that asap.

Please, tell us if you advanced with the issue.
Thanks,

**Comment by aherrick** [comment]
Thanks for the response.

Here is what my SDK looks like and it's still not working.

![3-8-2017 3-59-07 pm](https://cloud.githubusercontent.com/assets/586671/23723796/8145d23e-0418-11e7-9173-16de5c13b6cf.png)

The error I'm getting is interesting:

`
C:\eShopOnContainers\src\Mobile\eShopOnContainers\eShopOnContainers.Droid\eShopOnContainers.Droid.csproj : error  : The imported project "C:\eShopOnContainers\packages\Xamarin.Android.Support.Vector.Drawable.23.3.0\build\Xamarin.Android.Support.Vector.Drawable.targets" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.  C:\eShopOnContainers\src\Mobile\eShopOnContainers\eShopOnContainers.Droid\eShopOnContainers.Droid.csproj
`

I'm not sure why it's trying to look in this path:
`C:\eShopOnContainers\packages`

When this is the actual path:
`
C:\eShopOnContainers\src\Mobile\packages`

I'm opening this SLN file:
`C:\eShopOnContainers\src\Mobile\eShopOnContainers.Xamarin.sln`

**Comment by eiximenis** [comment]
Hi @aherrick!

Try to open the eShopOnContainers-MobileApps.sln located in the root of the repo. I think the issue is caused by opening this other solution file. Nuget packages are restored in the same path as the sln and csproj stores relative routes to them. I checked the csproj and they store the routes to a packages folder located in the root of repo.

If this solves your issue, I think we should delete the two slns located in /src/Mobile.

**Comment by CESARDELATORRE** [comment]
Let's keep just the solutions at the root level so we don't need to maintain other solutions.
So, yes, let's delete the internal .sln located within the /src/Mobile, ok?

**Comment by aherrick** [comment]
Looks like that got it... now I'm able to deploy. I'd say this can be closed. Thanks!

**Comment by CESARDELATORRE** [comment]
@aherrick  Cool! - Btw, if you deploy to a real device (Android/iOS/Windows), then, no issues.
But if you deploy and run the end to end app on an Android emulator, take into account the incompatibility between Hyper-V that needs "Docker for Windows" and the Intel HAXM virtualization system that comes with Google's Android Emulator.

You can also run the Xamarin app "alone" with no Docker containers/services because we implemented mocks in the Xamarin app, so you should be able to run it with fake data without accessing the real services, as well.

I mentioned this issue in the setup instructions:
https://github.com/dotnet/eShopOnContainers/wiki/02.-Setting-eShopOnContainer-solution-up-in-a-Visual-Studio-2017-environment 

IMPORTANT: Check that you don't have any other hypervisor installed that might be not compatible with Hyper-V. For instance, Intel HAXM can be installed by VS 2017 if you chose to install Google's Android emulator which works on top of Intel HAXM. In that case, you'd need to uninstall Google's Android emulator and Intel HAXM. VS 2017 recommends to install the Google Android emulator because it is the only Android emulator with support for Google Play Store, Google Maps, etc. However, take into account that it currently is not compatible with Hyper-V, so you might have incompatibilities with this scenario.

**Comment by CESARDELATORRE** [comment]
Closing this issue. Thanks Eduard! :)
